src/Console/ServerBlock.php

```diff
@@ -87,17 +91,77 @@ protected function doExecute()
 				return $this->addBlockedServer($this->config);
 			case 'remove':
 				return $this->removeBlockedServer($this->config);
+			case 'export':
+				return $this->exportBlockedServers($this->config);
+			case 'import':
+				return $this->importBlockedServers($this->config);
 			default:
 				throw new CommandArgsException('Unknown command.');
 				break;
 		}
 	}
 
 	/**
-	 * Prints the whole list of blocked domains including the reason
+	 * Exports the list of blocked domains including the reason for the
+	 * block to a CSV file.
+	 *
+	 * @param IConfig $config
+	 */
+	private function exportBlockedServers(IConfig $config)
+	{
+		$filename = $this->getArgument(1);
+		$blocklist = $config->get('system', 'blocklist', []);
+		$fp = fopen($filename, 'w');
```

Comment:
Please check that the file was correctly open and output and error if it wasn't.